crates/task-impls/src/da.rs

```diff
@@ -371,18 +370,17 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
 
                 return None;
             }
-            HotShotEvent::BlockReady(payload, metadata, view) => {
+            HotShotEvent::BlockReady(encoded_transactions, metadata, view) => {
                 self.da_network
                     .inject_consensus_info(ConsensusIntentEvent::CancelPollForTransactions(*view))
                     .await;
 
-                let payload_commitment = payload.commit();
+                let payload_commitment = vid_commitment(encoded_transactions.clone());
```

Comment:
Maybe we will fix this in the other VID-related tasks(?) but I don't think we should be computing the VID commitment in the DA leader task at all. It's unnecessary, because the leader is already going to compute the commitment in the VID task. So maybe it should be the VID task, not the DA task, that sends `SendPayloadCommitmentAndMetadata`

Reply:
We have an in-progress PR by @rob-maron to split the `BlockReady` event but `SendPayloadCommitmentAndMetadata` isn't in the plan. I think this is a good idea to move it to the VID task! I'll do it in this PR then.